File: python/916. Word Subsets.py
class Solution:
    def wordSubsets(self, words1: List[str], words2: List[str]) -> List[str]:
        freq = [[Counter(word).get(chr(i + ord('a')), 0) for i in range(26)] for word in words1]
        req_freq = [max(Counter(word).get(chr(i + ord('a')), 0) for word in words2) for i in range(26)]
        answer = []
        for i, word in enumerate(words1):
            if all(freq[i][j] >= req_freq[j] for j in range(26)):
                answer.append(word)
        return answer

File: java/916. Word Subsets.java
class Solution {
    public List<String> wordSubsets(String[] words1, String[] words2) {
        int[] reqFreq = new int[26];
        for (String w : words2) {
            int[] freq = getFreq(w);
            for (int i = 0; i < 26; i++) {
                reqFreq[i] = Math.max(reqFreq[i], freq[i]);
            }
        }
        List<String> answer = new ArrayList<>();
        OUTER: for (String w : words1) {
            int[] freq = getFreq(w);
            for (int i = 0; i < 26; i++) {
                if (freq[i] < reqFreq[i]) {
                    continue OUTER;
                }
            }
            answer.add(w);
        }
        return answer;
    }
    private int[] getFreq(String w) {
        int[] freq = new int[26];
        for (char c : w.toCharArray()) {
            freq[c - 'a']++;
        }
        return freq;
    }
}